PowerEditor/src/WinControls/Preference/preferenceDlg.cpp

```diff
@@ -1133,7 +1213,8 @@ intptr_t CALLBACK DarkModeSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPARA
 	{
 		case WM_INITDIALOG:
 		{
-			::SendDlgItemMessage(_hSelf, IDC_CHECK_DARKMODE_ENABLE, BM_SETCHECK, nppGUI._darkmode._isEnabled, 0);
+			const int topControlID = NppDarkMode::isWindowsModeEnabled() ? IDC_RADIO_DARKMODE_FOLLOWWINDOWS : NppDarkMode::isEnabled() ? IDC_RADIO_DARKMODE_DARKMODE : IDC_RADIO_DARKMODE_LIGHTMODE;
+			::SendDlgItemMessage(_hSelf, topControlID, BM_SETCHECK, BST_CHECKED, 0);
 

```

Comment:
@donho 
should we disable button for older OS?
`::EnableWindow(::GetDlgItem(_hSelf, IDC_RADIO_DARKMODE_FOLLOWWINDOWS), NppDarkMode::isWindows10());`

Reply:
@ozone10 
> should we disable button for older OS?

By "older OS" you mean Windows 8 and before right?

Reply:
I also modified `const int topControlID = NppDarkMode::isWindowsModeEnabled() ? IDC_RADIO_DARKMODE_FOLLOWWINDOWS : NppDarkMode::isEnabled() ? IDC_RADIO_DARKMODE_DARKMODE : IDC_RADIO_DARKMODE_LIGHTMODE;` to `const int topControlID = NppDarkMode::isWindowsModeEnabled() && !NppDarkMode::isWindows10() ? IDC_RADIO_DARKMODE_FOLLOWWINDOWS : NppDarkMode::isEnabled() ? IDC_RADIO_DARKMODE_DARKMODE : IDC_RADIO_DARKMODE_LIGHTMODE;`

Reply:
@donho 
> By "older OS" you mean Windows 8 and before right?

yes